app/models/concerns/twitter_user/api.rb

```diff
@@ -0,0 +1,279 @@
+require 'active_support/concern'
+
+module Concerns::TwitterUser::Api
+  extend ActiveSupport::Concern
+
+  included do
+    def dummy_client
+      @dummy_client ||= ExTwitter::Client.new
+    end
+
+    def one_sided_following
+      dummy_client.one_sided_following(self)
+    end
+
+    def one_sided_followers
+      dummy_client.one_sided_followers(self)
+    end
+
+    def mutual_friends
+      dummy_client.mutual_friends(self)
+    end
+
+    def common_friends(other)
+      return [] if other.blank?
+      dummy_client.common_friends(self, other)
+    end
+
+    def common_followers(other)
+      return [] if other.blank?
+      dummy_client.common_followers(self, other)
+    end
+
+    def removing
+      return [] unless TwitterUser.has_more_than_two_records?(uid, user_id)
+      TwitterUser.where(uid: uid, user_id: user_id).order(created_at: :asc).each_cons(2).map do |old_one, new_one|
+        dummy_client.removing(old_one, new_one)
+      end.flatten
+    end
+
+    def removed
+      return [] unless TwitterUser.has_more_than_two_records?(uid, user_id)
+      TwitterUser.where(uid: uid, user_id: user_id).order(created_at: :asc).each_cons(2).map do |old_one, new_one|
+        dummy_client.removed(old_one, new_one)
+      end.flatten
+    end
+
+    def replying(options = {})
+      if statuses.any?
+        client.replying(statuses.to_a, options)
+      else
+        client.replying(uid.to_i, options)
+      end.map { |u| u.uid = u.id; u }
+    end
+
+    def replied(options = {})
+      result =
+        if ego_surfing?
+          if mentions.any?
+            mentions.map { |m| m.user }.map { |u| u.uid = u.id; u }
+          else
+            client.replied(uid.to_i, options)
+          end
+        elsif search_results.any?
+          uids = dummy_client._extract_uids(search_results.to_a)
+          dummy_client._extract_users(search_results.to_a, uids)
+        else
+          client.replied(uid.to_i, options)
+        end.map { |u| u.uid = u.id; u }
+
+      (options.has_key?(:uniq) && !options[:uniq]) ? result : result.uniq { |u| u.uid.to_i }
+    end
+
+    def favoriting(options = {})
+      if favorites.any?
+        client.favoriting(favorites.to_a, options)
+      else
+        client.favoriting(uid.to_i, options)
+      end.map { |u| u.uid = u.id; u }
+    end
+
+    def inactive_friends
+      dummy_client._extract_inactive_users(friends, authorized: ego_surfing?)
+    end
+
+    def inactive_followers
+      dummy_client._extract_inactive_users(followers, authorized: ego_surfing?)
+    end
+
+    def clusters_belong_to
+      text = statuses.map{|s| s.text }.join(' ')
+      dummy_client.clusters_belong_to(text)
+    end
+
+    def close_friends(options = {})
+      min = options.has_key?(:min) ? options.delete(:min) : 1
+      options.merge!(uniq: false, min: min)
```

Comment:
__[rubocop]__
* Use `options[:uniq] = false; options[:min] = min` instead of `options.merge!(uniq: false, min: min)`. [:ref](https://github.com/JuanitoFatas/fast-ruby#hashmerge-vs-hash-code)